mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -347,4 +304,126 @@ private CqlNameGenerator buildCqlNameGenerator(TypeElement classElement) {
+
+    Transient transientAnnotation = (Transient) annotations.get(Transient.class);
+    // check if property name is included in @TransientProperties
+    // -or- if property is annotated with @Transient
+    // -or- if field has transient keyword modifier
+    boolean isTransient =
+        transientProperties.contains(propertyName)
+            || transientAnnotation != null
+            || field != null && field.getModifiers().contains(Modifier.TRANSIENT);
+
+    // if annotations contains an exclusive annotation that isn't transient, raise
+    // an error here.
+    Class<? extends Annotation> exclusiveAnnotation = getExclusiveAnnotation(annotations);
+    if (isTransient && transientAnnotation == null && exclusiveAnnotation != null) {
+      Element element = field != null ? field : getMethod;
+      context
+          .getMessager()
+          .error(
+              element,
+              "Property that is considered transient cannot be annotated with @%s.",
+              exclusiveAnnotation.getSimpleName());
+    }
+
+    return isTransient;
+  }
+
+  private Set<String> getTransientPropertyNames(TypeElement classElement) {
+    TransientProperties transientProperties = classElement.getAnnotation(TransientProperties.class);
+
+    return transientProperties != null
+        ? Sets.newHashSet(transientProperties.value())
+        : Collections.emptySet();
+  }
+
+  private void reportMultipleAnnotationError(
+      Element element, Class<? extends Annotation> a0, Class<? extends Annotation> a1) {
+    if (a0 == a1) {
+      context
+          .getMessager()
+          .warn(
+              element,
+              "@%s should be used either on the field or the getter, but not both. "
+                  + "The annotation on this field will be ignored.",
+              a0.getSimpleName());
+    } else {
+      context
+          .getMessager()
+          .error(
+              element,
+              "Properties can't be annotated with both @%s and @%s.",
+              a0.getSimpleName(),
+              a1.getSimpleName());
+    }
+  }
+
+  private Map<Class<? extends Annotation>, Annotation> scanPropertyAnnotations(
+      ExecutableElement getMethod, VariableElement field) {
+    Map<Class<? extends Annotation>, Annotation> annotations = Maps.newHashMap();
+
+    // scan methods first as they should take precedence.
+    scanMethodAnnotations(getMethod, annotations, null);
+    if (field != null) {
+      scanFieldAnnotations(field, annotations, getExclusiveAnnotation(annotations));
+    }
+
+    return ImmutableMap.copyOf(annotations);
+  }
+
+  private Class<? extends Annotation> getExclusiveAnnotation(
+      Map<Class<? extends Annotation>, Annotation> annotations) {
+    for (Class<? extends Annotation> annotationClass : annotations.keySet()) {
+      if (EXCLUSIVE_PROPERTY_ANNOTATIONS.contains(annotationClass)) {
+        return annotationClass;
+      }
+    }
+    return null;
+  }
+
+  private void scanFieldAnnotations(
```

Comment:
code for `scanFieldAnnotations` and `scanMethodAnnotations` is largely the same, but anticipate that changing with [JAVA-2111](https://datastax-oss.atlassian.net/browse/JAVA-2111)